rakefile

```diff
@@ -1,13 +1,85 @@
 require 'rototiller'
 require 'rspec/core/rake_task'
+require 'net/http'
+require 'timeout'
+require 'json'
 
 task :default => :performance
 
-desc 'Run Performance setup'
-rototiller_task :performance do |t|
+desc 'Provision systems using ABS (or use already set ABS_RESOURCE_HOSTS) and then execute beaker performance setup and tests'
+rototiller_task :performance do
+  Rake::Task["performance_provision_with_abs"].execute unless ENV['ABS_RESOURCE_HOSTS']
+  Rake::Task["performance_without_provision"].execute
+end
+
+desc 'Provision systems for the performance task with ABS'
+rototiller_task :performance_provision_with_abs do
+
+  ENV['ABS_JOB_ID'] = rand(100000..999999) unless ['ABS_JOB_ID']
+  uri = URI('https://cinext-abs-test.delivery.puppetlabs.net/api/v2/request')
+  req = Net::HTTP::Post.new(uri, 'Content-Type' => 'application/json', )
+  req.body = {'resources': {'centos-7-x86-64-west': 2},'job': { 'id': "#{ENV['ABS_JOB_ID']}", 'tags': {'user': 'jenkins','jenkins_build_url': 'https://jenkins/job/local-perf-test'}}}.to_json
+  http = Net::HTTP.new(uri.host, uri.port)
+  http.use_ssl = true
+
+  puts "sending request body: #{req.body}"
+  puts "to uri: #{uri}"
+  res = nil
+
+  # Wait up to 5 minutes for a response from ABS with provisioned hosts
+  status = Timeout::timeout(300) do
+    while res.nil? || res.body == ''
+      begin
+        res = http.request(req)
+      rescue => ex
+        puts ex
+        puts ex.backtrace
+      end
+      if res.nil? || res.body == ''
+        puts 'Awaiting ABS_RESOURCE_HOSTS response from ABS...'
+        sleep(10)
+      end
+    end
+  end
+  ENV['ABS_RESOURCE_HOSTS'] = res.body
+  puts "ABS_RESOURCE_HOSTS=#{ENV['ABS_RESOURCE_HOSTS']}"
+end
+
+rototiller_task :performance_deprovision_with_abs do
+  raise 'performance_deprovision_with_abs task requires values for env variables: ABS_RESOURCE_HOSTS and ABS_JOB_ID' unless ENV['ABS_RESOURCE_HOSTS'] && ENV['ABS_JOB_ID']
```

Comment:
do we do this currently due to deficiencies in rototiller's required env vars?  there is very likely nothing to handle this exception.

Reply:
Oh, the check for BEAKER_INSTALL_TYPE makes sense because some of the other env vars are optional depending on what it is set to, but you're right, there's no need for this check.

Reply:
yeah, there is a ticket for pattern matching on env var values for tiller.  but i think that thing needs empty env var and other edge cases of the env var interface to be cleared up first anyway